custom_components/rivian/__init__.py

```diff
@@ -192,6 +195,13 @@ async def _update_api_data(self):
             self._login_attempts = 0
 
             vehicle_info_items = self.build_vehicle_info_dict(vijson)
+
+            if self._wallboxes or self._wallboxes is None:
```

Comment:
Wouldn't this always be true?

Reply:
Though untested, if someone doesn't have a wallbox, I "think" the query returns an empty list i.e. `[]`. This would cause it to evaluate to false and not run.